content/improving-dora-metrics.md

```diff
@@ -0,0 +1,118 @@
+
+1. Are testing best practices consistently followed across the SDLC?
+2. Are test cases comprehensive enough to detect issues in lower environments?
+3. Are lower environments similar enough to production to allow for reliable test results? 
+
+Depending on how a rollback is initiated, Change Failure Rate can potentially be tracked at the repository level. This would require each hotfix/ remediation build to be labelled as such in (say) GitHub. However, if the rollback protocol is not managed via the deployment pipeline, or if the failure is remediated using a manual hotfix; service management tools like ServiceNow can help record and measure CFR. 
+  
+## 4.4 Mean Time To Recovery (MTTR)
+
+Mean time to Recovery or Mean time to Resolution is the average amount of time required to recover from a failure. For the sake of continuity; let’s define failure as explained in 4.3 (CFR). If the application requires high-availability and high-resiliency, MTTR becomes the most important metric to track and improve. Answering some of these questions could help identify barriers to improving MTTR:
+
+1. How are issues (errors/ defects/ outages) detected today?
+2. How are incidents routed to the appropriate resolution teams?
+3. Do development teams have expected Service Levels objectives and indicators codified?
+4. Are resolution teams aware of the mitigation protocol?
+5. Are resolution teams empowered to resolve issues and make quick decisions?
+
+Similar to Change Failure Rare, MTTR can be complex to measure. At the risk of oversimplifying this nuanced process, teams may rely on monitoring and observability platforms to capture the start and end times of an incident. Depending on the nature of the issues; Application Insights within Azure Monitor or even PagerDuty can be leveraged to measure MTTR. 
+
+---
+
+# 5.0 Prioritize and eliminate barriers
+
+As a proponent of a data-driven decision making culture, I have avoided prescriptive approaches to improving DORA metrics. It is best to contemplate the barriers on a team level, or better still, an application level; and focus on dismantling these barriers in a methodical and purposeful way guided by priority and return on investment. 
+
+That said, most of the identified barriers can likely be classified into the following categories (in order of difficulty):
+
+1. Skill/Knowledge constraints
+2. Process constraints
+3. Budgetary constraints
+4. Architecture/ Infrastructure limitations
+
+---
+
+# 6.0 Conclusion
+
+It is likely that some of the barriers we encountered can be eliminated by coaching team members on existing organizational protocols and widely accepted engineering best practices around development, testing, deployment, monitoring, error detection and mitigation. Communities of Practice must make contin
```

Comment:
Communities of Practice must make continuous learning a key priority, and time must be allocated for learning and teaching.

Reply:
Looks good to me